main.cpp

```diff
@@ -0,0 +1,23 @@
+#include <iostream>
+#include <array>
+
+#include "prime.h"
+
+int main()
+{
+    int input_num = 0;
+    std::array<int, ARRAY_LEN> array_numbers {};

```

Comment:
Great use of STL! Unfortunately here this isn't correct since this is compile-time, even though size is only used at run-time. Also, do not use `std::vector` but instead use dynamic memory allocation